### PR TITLE
fix(button): remove role="button" from button styled anchors 

### LINKED ISSUE #5112: <Button href="xyz"> missing label
## Environment

Chrome Mac OS X
DAP chrome plugin
Jan2020 archive, IBM Accessibility January 2020 ruleset

## Detailed description

> What version of the Carbon Design System are you using?

Carbon Components React 7.9.2

> What WCAG 2.1 checkpoint does the issue violate?

1.3.1 & 4.1.2

## Steps to reproduce the issue

https://codesandbox.io/s/dreamy-goldberg-z2miy

## Additional information

image

### LINKED ISSUE #5521: Default button has 3 DAP errors
For non `button` element buttons the implicit labelling is not being picked up by DAP

2020-03-03 15_38_58-Storybook
<!-- Feel free to remove sections that aren't relevant.

### PR DESCRIPTION
**Pull request by dakahn**
Closes #5521 
Closes #5112 

Implicit labeling (having wrapped text in the sub-tree acting as the label) only works for native `button` elements. ~~For instances where the button is an anchor tag, p tag or otherwise we need to specify the label with text matching the visible label.~~

#### Changelog

**New**

- ~~added an `aria-label: children` to the `anchorProps` object~~
- removed `role="button"` 

#### Testing / Reviewing

Since button is a widely used generic component I could totally be missing something/some use case. Otherwise just open up Chrome and run DAP with the latest ruleset and you should see zero errors reported 🏄

### CODE REVIEW

**Comment by asudoh** [comment]
A question; What happens if the button has non-text content?

**Comment by dakahn** [comment]
@asudoh Well, if there was no label whatsoever DAP would throw and error for an anchor tag not having an associated label. Is that a use case we support? For an icon I thought we set a label on the svg.

2020-03-04 10_25_52-Storybook

**Review by abbeyhrt — APPROVED** [review]
Looks good to me! Separately, I see that DAP is saying that `p` and `a` are form control elements, are they really? Or am I misunderstanding and it's just picking up the `role='button'`?

**Comment by dakahn** [comment]
@abbeyhrt right yeah! They become form control elements the moment we designate them `role="button"`

**Comment by joshblack** [comment]
Wow, definitely an awkward violation lol. Would you want to add a quick test to check for this too with any links to refer to why we need to do this?

**Comment by dakahn** [comment]
@joshblack added test! Though you'll notice we we had two nested divs inside our test button that I removed -- they didn't seem to be directly related to any of the tests I saw in that unit, but there might be something I'm missing.

File: packages/react/src/components/Button/Button-test.js [hunk 1]
```diff
@@ -99,6 +98,17 @@ describe('Button', () => {
...
+
+    /**
+     * Implicit labeling only works for native button elements. We have
+     * to designate an `aria-label` that matches the visible text to avoid
+     * DAP violations.
+     *
+     * ref: https://github.com/carbon-design-system/carbon/issues/5521
+     */
+    it('should have an aria-label matching visible label text', () => {
+      expect(wrapper.props()['aria-label']).toEqual('Child');
```
Comment (joshblack) [inline packages/react/src/components/Button/Button-test.js]:
```suggestion
      expect(wrapper.prop('aria-label').toEqual('Child');
```

**Review by joshblack — CHANGES_REQUESTED** [review]

**Comment by asudoh** [comment]
@dakahn You could detect if `children` is a string. If it's the case you can still propagate it to `aria-label`. Otherwise we can let application to populate `aria-label`.

**Comment by joshblack** [comment]
That's a great point ☝️, `children` is of type `PropTypes.node` so may not contain text. Would we instead need a prop type warning such that `aria-label` is required if `href` is supplied?

**Comment by dakahn** [comment]
Hey @joshblack brought up a good point that it probably makes more sense to simply remove `role="button"` for any anchors or whatever other non `<button>` elements we're only styling like buttons. 👍

**Comment by abbeyhrt** [comment]
@dakahn I'm not really sure how the non-button elements styled to look like buttons are used but is it a problem that they're no longer announced as buttons?

**Comment by tw15egan** [comment]
@abbeyhrt my understanding is they are used when you are navigating to different parts of an app, so I think it would make more sense that they were _not_ announced as buttons. I believe they are just styled like them since a lone link doesn't seem to have the same affordance/CTA as a button.

**Comment by abbeyhrt** [comment]
@tw15egan Ah gotcha, makes sense! Thanks for explaining!

**Review by tw15egan — APPROVED** [review]
Makes sense to me, DAP errors have been eliminated ☠️ 

 👍 ✅

**Review by joshblack — APPROVED** [review]

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @dakahn!

**Pull request merged**